### ISSUE TITLE
SwipeView control throws a System.AccessViolationException in Windows.

### ISSUE THREAD

**Issue by kg8z** [MoneyFox/MoneyFox#2596]
**Describe the bug**
The SwipeView control was added to the following views to allow Edit/Delete operations to be performed with a right-swipe on touch enabled devices:

- Src/MoneyFox.Ui/Views/Accounts/AccountListPage.xaml
- Src/MoneyFox.Ui/Views/Categories/CategoryListPage.xaml
- Src/MoneyFox.Ui/Views/Categories/SelectedCategoryPage.xaml
- Src/MoneyFox.Ui/Views/Popups/CategorySelectionPopup.xaml

While it works properly on iOS and Android, the SwipeView control throws a System.AccessViolationException exception with the message "Attempted to read or write protected memory. This is often an indication that other memory is corrupt." when the user interacts with the control in Windows. This appears to be a bug in the .NET code.

Removing the SwipeView control from these 4 files mitigates the exception and restores the pre-MAUI behavior of Edit/Delete operations being performed within the subsequent edit view. I would suggest we remove the new swipe functionality until the bug in SwipeView is resolved as there is no way to easily retain it for mobile but not desktop.

**To Reproduce**
1. Run the app on Windows (I'm using Windows 11 Home)
2. Click on the _Add Payment_ button
3. Click on the _Select category.._ button
4. An exception will be thrown

**Expected behavior**
Clicking the button to open the category list should present the form, not throw an exception.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11 Home
 - Version	10.0.22621 Build 22621

**Additional context**

**Comment by kg8z** [comment]
@NPadrutt - If you agree with my suggested solution (remove SwipeView control), I already have changes ready to fix this bug.

**Comment by NPadrutt** [comment]
I think the main question then would be how to access the edit settings for an account. A tap brings you to the payment view. A long tap would work, but that is not readily available in MAUI.

Also, I found this: https://github.com/dotnet/maui/issues/6153
Might there be an issue in our code still?

**Comment by kg8z** [comment]
@NPadrutt

Here's another reported issue I found that's consistent with the behavior I'm seeing:[dotnet/maui#6401](https://github.com/dotnet/maui/issues/6401). Both this and the one you referenced have been closed so I would have expected the fixes to be in the current runtimes.

I've tried tests that reduce the SwipeView control just down to formatting tags (removed all binding and just left a hard-coded Text element) and it's still throwing the same exception. This, combined with the fact the same code works perfectly on iOS and Android but not Windows, seems to strongly suggest the bug is in the control and not our code. I was going to try testing against the net6.0 framework to see if it was introduced in version 7.0 but the MAUI libraries don't work on the older framework.

You're right; I missed the fact there's currently no way to edit/delete an account without the SwipeView. Opening a Category allows values to be changed and exposes a Delete button, so this is fine. But Accounts don't offer that same functionality.

Do you have a Windows machine? If so, are you able to reproduce the problem (to make sure it's not isolated to my environment)?

Assuming there's a bug in the MAUI code, would you prefer we raise an issue and wait for it to be fixed (meaning the Windows app will continue to crash) or implement alternative Edit/Delete functions?

**Comment by NPadrutt** [comment]
I checked on my computer and I have the same. But I get this error as well when I create a new account or payment. So I think there is currently a more general issue with the windows version. I wonder if there is another bug here in maui..

**Comment by kg8z** [comment]
@NPadrutt - I suspect the errors you're getting on new account/payment creation are also because of the bug in the SwipeView control. Clicking the "Go to Accounts" link or aggregate Assets box to bring up the account list throws an exception because of the SwipeView control. Also, clicking the "Select category.." button when adding a new Payment (or modifying an existing one) throws an exception because of the SwipeView control. When I remove the SwipeView control from the 4 files mentioned above, all of these operations work successfully. If you're seeing the error with different steps, let me know and I can test whether it's also because of the SwipeView bug.

**Comment by NPadrutt** [comment]
Aah, I see.. is ot possible to hide or deactivate the swipe view on windows? Then we could see if we can implement a right click on windows. I think that would be the more intuitive way anyway.

Also, did you do a minimal repro? If so you could create a GitHub issue with the maui team so they can look at it.

**Comment by NPadrutt** [comment]
There seems to be an open issue for that https://github.com/dotnet/maui/issues/8870

**Comment by kg8z** [comment]
Ah yeah. Great find. A bit concerning that it was reported back in July but has yet to be fixed, though it's encouraging the comment stream is active. I was considering workarounds but as someone mentioned in the bug you referenced, "There's no conditional compilation in XAML, so the only other thing I can think of is having a separate page / view for Windows vs other platforms." I was hoping to avoid the need to split the form into a Windows version and mobile version but that seems to be the only viable option if we want to retain the SlideView for mobile. Perhaps the bug will get some attention soon so we don't need to regress to different mobile/Windows UI code.

**Comment by NPadrutt** [comment]
yeah, I tried around with work arounds aswell without success. It is unfortunate but I think making a dedicated desktop page is the best way as of now. Given that we can get rid of the dedicated windows project instead that is still progress. But yeah there are currently quite a bit of bugs around unfortunately.
btw, instead of the swipe view we could just use the context menu. that was introduce in .net 7 for desktop platforms.

**Comment by NPadrutt** [comment]
For know I created a new desktop page for the category and the account list and removed the swipeview from there. especially for the account list we would have to add a context menu to have all functionality available.

**Comment by NPadrutt** [comment]
I keep this open to keep track of the .net maui bug.

**Comment by kg8z** [comment]
Got it. Too bad it required a separate page and defeats the benefits of MAUI, but makes sense as really the only viable way to fix the problem. Let's hope the fix the MAUI bug soon.
